components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/DefaultOIDCClaimsCallbackHandler.java

```diff
@@ -285,6 +285,12 @@ private Map<String, Object> getUserConsentedClaims(Map<String, Object> userClaim
     private Map<ClaimMapping, String> getCachedUserAttributes(OAuthTokenReqMessageContext requestMsgCtx) {
 
         Map<ClaimMapping, String> userAttributes = getUserAttributesCachedAgainstToken(getAccessToken(requestMsgCtx));
+        Map<ClaimMapping, String> userAttributesFromAuthorizedUser = requestMsgCtx.getAuthorizedUser().
+                getUserAttributes();
+        if (!userAttributes.isEmpty() && !userAttributes.equals(userAttributesFromAuthorizedUser)) {
```

Comment:
is checking object equals the expectation here?

Reply:
Yes, checking the map object equal